returnn/torch/engine.py

```diff
@@ -186,6 +187,14 @@ def train_epoch(self):
                     for name, loss in ctx_losses_dict.items()
                 }
             )
+            
+            if self.config.bool("stop_on_nonfinite_train_score", True):
```

Comment:
I would move this to a separate function to not make the `train_epoch` function too large.